prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -154,36 +165,83 @@ object FhirPathUtils : Logging {
      * @return the converted HL7 DTM
      */
     fun convertDateTimeToHL7(dateTime: BaseDateTimeType): String {
-        val offsetDateTime: OffsetDateTime = OffsetDateTime.ofInstant(
-            dateTime.value.toInstant(), dateTime.timeZone.toZoneId()
-        )
+        /**
+         * Set the timezone for an [hl7DateTime] if a timezone was specified.
+         * @return the updated [hl7DateTime] object
+         */
+        fun setTimezone(hl7DateTime: DTM): DTM {
+            dateTime.timeZone?.let {
+                // This is strange way to set the timezone offset, but it is an integer with the leftmost two digits as the hour
+                // and the rightmost two digits as minutes (e.g. -0400)
+                val hour = dateTime.timeZone.rawOffset / 1000 / 60 / 60
+                val min = dateTime.timeZone.rawOffset / 1000 / 60 % 60
+                hl7DateTime.setOffset(hour * 100 + min)
+            }
+            return hl7DateTime
+        }
 
-        return offsetDateTime.format(dateTimeFormatter)
+        val hl7DateTime = DTM(null)
+
+        return when (dateTime.precision) {
+            // HL7 Date time does not support only year, or year and month
```

Comment:
I'm confused by this comment the HL7 DTM says it supports just year by itself in the spec - https://hl7-definition.caristix.com/v2/HL7v2.6/DataTypes/DTM

Reply:
Old comment.  Needs to be removed